### PR TITLE
Scene Menu and Dungeon flags

### PR DESCRIPTION
**Pull request by Macrepeh**
Added scene menu with a (currently non-functional) wind direction modifier and time/date viewer.
Added dungeon flags to flag menu. Currently includes keys, map, boss key, compass, and — for ET/WT — a dungeon partner toggle. Additional features to both menus listed may be added at a later point if certain requisites are fulfilled.

### CODE REVIEW

File: external/libtww/d/save/d_save.h [hunk 1]
```diff
@@ -426,11 +427,15 @@ static_assert(sizeof(dSv_player_c) == 0x380);
 
 class dSv_memBit_c {
 public:
+    void setKeyNum(int num) { mKeyNum = num; }
+    u8& getKeyNum() { return mKeyNum; }
+    void offDungeonItem(int flag) { mDungeonItem &= ~(1 << flag); }
+ //   void onDungeonSwitch(int flag) { mSwitch[flag >> 5] = mSwitch[flag >> 5] | 1 << (flag & 0x1F); }
```
Comment (samrg97) [inline external/libtww/d/save/d_save.h:433]: Is this code you don't want anymore?

File: include/menu.h [hunk 2]
```diff
@@ -38,6 +39,10 @@ enum {
     GZ_KORL_FLAG_MENU,
     GZ_SAVEWARP_FLAG_MENU,
 
+    /* Scene sub-menus */
+    GZ_DUNGEON_MENU,
+ //   GZ_PLACEHOLDER2_MENU,
```
Comment (samrg97) [inline include/menu.h]: Why is this here? Does this affect your flag-adding script?

File: include/menus/scene_menu.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#include "libtww/d/kankyo/d_kankyo.h"
+#include "libtww/d/com/d_com_inf_game.h"
+#include "libtww/f_op/f_op_scene_req.h"
+#include "libtww/m_Do/m_Do_main.h"
+#include "controller.h"
+
+enum SceneMenuIndex {
+    MODIFY_WIND_INDEX,
+    TIME_DISP_INDEX,
+ //   DUNGEON_INDEX,
```
Comment (samrg97) [inline include/menus/scene_menu.h]: Dungeon stuff is under flags, right? Then this should be deleted

File: source/menus/dungeon_menu.cpp [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        cursor.move(5, MAX_MOVE);
+
+        if (cursor.y == PICK_DUNGEON_INDEX) {
+            l_selDun = cursor.x;
+        }
+        l_keyNum = dComIfGs_getKeyNum(area_id);
+        selectDungeon();
+        break;
+    case MODIFY_KEYS_INDEX:
+        cursor.move(0, LINE_NUM);
```
Comment (samrg97) [inline source/menus/dungeon_menu.cpp]: Should this say "MAX_MOVE" instead of "LINE_NUM"?

File: source/menus/dungeon_menu.cpp [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            }
+        }
+    }
+    dComIfGs_getSave(g_dComIfG_gameInfo.info.mDan.mStageNo);
+    tww_sprintf(lines[PICK_DUNGEON_INDEX].value, " <%s>", get_dun_name(area_id));
+    tww_sprintf(lines[MODIFY_KEYS_INDEX].value, " <%d>", dComIfGs_getKeyNum(area_id));
+
+    if ((area_id != 6) && (area_id != 7)) {
+        GZ_drawMenuLines(lines, cursor.y, LINE_NUM - 1);
+        if (cursor.y != PICK_DUNGEON_INDEX) {
```
Comment (samrg97) [inline source/menus/dungeon_menu.cpp]: Does this if statement do anything?

**Review by samrg97 — COMMENTED** [review]
Can anyone see this?

File: source/menus/dungeon_menu.cpp (hunk 5 above)
Comment (Macrepeh) [reply source/menus/dungeon_menu.cpp]: It did, but not any more

File: source/menus/dungeon_menu.cpp (hunk 4 above)
Comment (Macrepeh) [reply source/menus/dungeon_menu.cpp]: It doesn't seem to matter either way, but I can change it for consistency

File: include/menus/scene_menu.h (hunk 3 above)
Comment (Macrepeh) [reply include/menus/scene_menu.h]: It should be

File: include/menu.h (hunk 2 above)
Comment (Macrepeh) [reply include/menu.h]: the dungeon menu was once in the sceenes menu. This is a leftover. Now I understand why dark link is still called torch2, it is annoying going back and remembering to change all these little things but i will now.

File: external/libtww/d/save/d_save.h (hunk 1 above)
Comment (Macrepeh) [reply external/libtww/d/save/d_save.h:433]: This is code that does not currently do what it should, possibly due to memory being misaligned or some other problem I am unaware of. It is commented out for now to reflect the fact that it does not work.
